src/test/groovy/org/prebid/server/functional/util/privacy/gpp/GppConsent.groovy

```diff
@@ -0,0 +1,69 @@
+package org.prebid.server.functional.util.privacy.gpp
+
+import com.iab.gpp.encoder.GppModel
+import com.iab.gpp.encoder.section.TcfEuV2
+import com.iab.gpp.encoder.section.UspV1
+import org.prebid.server.functional.util.privacy.ConsentString
+
+import java.time.ZoneId
+import java.time.ZonedDateTime
+
+abstract class GppConsent implements ConsentString {
+
+    protected final gppModel = new GppModel()
+
+    protected GppConsent(Regions regions, Map<String, Object> fieldValues) {
+        fieldValues.each { fieldName, fieldValue ->
+            this.gppModel.setFieldValue(regions.value, fieldName, fieldValue)
+        }
+    }
+
+    @Override
+    String getConsentString() {
+        this.gppModel?.encode()
+    }
+
+    @Override
+    String toString() {
+        consentString
+    }
+
+    protected abstract static class Builder {
+
+        protected Regions regions
+        protected def fieldValues = [:]
+
+        Builder(Regions regions) {
+            this.regions = regions
+            fieldValues.Created = ZonedDateTime.now(ZoneId.of("UTC"))
```

Comment:
When you'll create `UspV1Consent` it'll failed because in this region doesn't present this field.